rt/rs/client/src/main/java/org/apache/cxf/jaxrs/client/spec/InvocationBuilderImpl.java

```diff
@@ -253,10 +253,13 @@ public Builder header(String name, Object value) {
 
     @Override
     public Builder headers(MultivaluedMap<String, Object> headers) {
-        RuntimeDelegate rd = HttpUtils.getOtherRuntimeDelegate();
-        for (Map.Entry<String, List<Object>> entry : headers.entrySet()) {
-            for (Object value : entry.getValue()) {
-                doSetHeader(rd, entry.getKey(), value);
+        webClient.removeAllHeaders();
+        if (headers != null) {
```

Comment:
does the TCK check that `.headers(null)` should clear all headers?  Seems odd to me.

Reply:
I agree it is odd... I'm not sure if the TCK tests it or not, but the [javadoc](https://jax-rs.github.io/apidocs/2.1/javax/ws/rs/client/Invocation.Builder.html#headers-javax.ws.rs.core.MultivaluedMap- ) says that if the passed-in map is null, then all existing headers should be removed.